Document annotation nginx.ingress.kubernetes.io/configuration-snippet (#43950)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

When a IngressClass resource has this annotation set to `"true"`, new Ingress resource
without a class specified will be assigned this default class.

### nginx.ingress.kubernetes.io/configuration-snippet

Type: Annotation

Example: `nginx.ingress.kubernetes.io/configuration-snippet: "  more_set_headers \"Request-Id: $req_id\";\nmore_set_headers \"Example: 42\";\n"`

Used on: Ingress

With use of this annotation someone can add additional configuration to the
Nginx location as mentioned in example above. `configuration-snippet` annotation is disabled

> by default Since version 1.9.0. `allow-snippet-annotations.` has to be explicitly enabled to

Reviewer: ```suggestion
by default since version 1.9.0. `allow-snippet-annotations.` has to be explicitly enabled to
```

use this annotation.
Enabling it can be dangerous in multi-tenant cluster, as it can lead people with otherwise

> limited permissions being able to retrieve all secrets of cluster.

Reviewer: ```suggestion
limited permissions being able to retrieve all secrets in the cluster.
```

When a IngressClass resource has this annotation set to `"true"`, new Ingress resource
without a class specified will be assigned this default class.

### nginx.ingress.kubernetes.io/configuration-snippet

Type: Annotation

Example: `nginx.ingress.kubernetes.io/configuration-snippet: |

>   more_set_headers "Request-Id: $req_id";`

Reviewer 2: Please avoid a multiline example here. I think this would be OK even if it's hard to read as it gives readers the sense that the annotation value is complex.
```suggestion
Example: `nginx.ingress.kubernetes.io/configuration-snippet: "  more_set_headers \"Request-Id: $req_id\";\nmore_set_headers \"Example: 42\";\n"`

Used on: Ingress

With use of this annotation someone can add additional configuration to the

> Nginx location as mentioned in example above. `configuration-snippet` annotation is disabled

Reviewer 2: I'd signpost readers to read the ingress controller's docs for the full details and an example.

When a IngressClass resource has this annotation set to `"true"`, new Ingress resource
without a class specified will be assigned this default class.

### nginx.ingress.kubernetes.io/configuration-snippet

Type: Annotation

Example: `nginx.ingress.kubernetes.io/configuration-snippet: "  more_set_headers \"Request-Id: $req_id\";\nmore_set_headers \"Example: 42\";\n"`

Used on: Ingress

> With use of this annotation someone can add additional 
> configuration to the
> Nginx location as mentioned in example above.

Reviewer 2: ```suggestion
You can use this annotation to set extra configuration on an Ingress that
uses the [NGINX Ingress controller](https://github.com/kubernetes/ingress-nginx/).
```

> `configuration-snippet` annotation is disabled

Reviewer 2: ```suggestion
The `configuration-snippet` annotation is ignored
```

> by default since version 1.9.0. `allow-snippet-annotations.` 

Reviewer 2: ```suggestion
by default since version 1.9.0 of the ingress controller.
The NGINX ingress controller setting `allow-snippet-annotations` 
```

has to be explicitly enabled to
use this annotation.

> Enabling it can be dangerous in multi-tenant cluster, as it can lead people with otherwise

Reviewer 2: ```suggestion
Enabling the annotation can be dangerous in a multi-tenant cluster, as it can lead people with otherwise
```

> limited permissions being able to retrieve all secrets in the cluster.

Reviewer 2: ```suggestion
limited permissions being able to retrieve all Secrets in the cluster.
```
